msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -201,21 +214,95 @@ public static void create(@NonNull final Context context,
+        if (context == null) {
+            throw new IllegalArgumentException("context is null.");
+        }
+
+        create(context,
+                loadConfiguration(context, configFileResourceId),
+                listener
+        );
+    }
+
+    /**
+     * {@link PublicClientApplication#createMultipleAccountPublicClientApplication(Context, int)} will read the client id and other configuration settings from the
+     * file included in your applications resources.
+     * <p>
+     * For more information on adding configuration files to your applications resources please
+     *
+     * @param context              Application's {@link Context}. The sdk requires the application context to be passed in
+     *                             {@link PublicClientApplication}. Cannot be null.
+     *                             <p>
+     *                             Note: The {@link Context} should be the application context instead of the running activity's context, which could potentially make the sdk hold a
+     *                             strong reference to the activity, thus preventing correct garbage collection and causing bugs.
+     *                             </p>
+     * @param configFileResourceId The resource ID of the raw file containing the JSON configuration for the PublicClientApplication
+     * @return An instance of IMultiAccountPublicClientApplication
+     * @see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
+     * @see <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
+     */
+    @WorkerThread
+    public static IMultipleAccountPublicClientApplication createMultipleAccountPublicClientApplication(@NonNull final Context context,
+                                                                                                       final int configFileResourceId) throws MsalException, InterruptedException {
+        if (context == null) {
+            throw new IllegalArgumentException("context is null.");
+        }
+        final PublicClientApplicationConfiguration configuration = loadConfiguration(context, configFileResourceId);
+
+        IPublicClientApplication application = createPublicClientApplication(context, configuration);
+        if(application instanceof IMultipleAccountPublicClientApplication){
+            return (IMultipleAccountPublicClientApplication)application;
+        }else{
+            if(configuration.mAccountMode == AccountMode.MULTIPLE && application.getIsSharedDevice())  {
+                throw new MsalClientException("AccountMode in configuration is set to multiple; however the device is marked as shared");
+            }
+            if(configuration.mAccountMode != AccountMode.MULTIPLE){
+                throw new MsalClientException("AccountMode in configuration is not set to multiple");
+            }
+            return null;
```

Comment:
Should this be another exception instead of returning `null`?

Reply:
+1

Reply:
fixed.